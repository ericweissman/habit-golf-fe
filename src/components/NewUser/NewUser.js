import React, { Component } from 'react';

export class NewUser extends Component {
  state = {
    name: '',
    email: '',
    password: '',
    confirmPassword: '',
  }

  handleChange = (e) => {
    const { name, value } = e.target;
    this.setState({ [name]: value })
  }

  render() {
    const { name, email, password, confirmPassword } = this.state
    return (
      <form className="new-user-form">
        <label>
          name:
            <input required type="text" name="name" value={name} onChange={this.handleChange} />
        </label>
        <label>
          email:
            <input required type="email" name="email" value={email} onChange={this.handleChange} />
        </label>
        <label>
          password:
            <input required type="password" name="password" value={password} onChange={this.handleChange} />
        </label>
        <label>
          confirm Password:
            <input required type="password" name="confirmPassword" value={confirmPassword} onChange={this.handleChange} />
        </label>
        <button>register</button>
      </form>
    );
  }
}

export default NewUser;
