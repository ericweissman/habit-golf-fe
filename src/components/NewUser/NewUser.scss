.new-user-form {
  @extend .sign-in-form;
  margin-top: 15px;
  label {
    padding: 0px 15px;
  }
  input {
    margin-top: 5px;
  }
  button {
    @extend .sign-in-btn;
    height: 50px;
    margin-bottom: 15px;
  }
}