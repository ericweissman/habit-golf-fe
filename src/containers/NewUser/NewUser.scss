.new-user-form {
  display: flex;
  flex-direction: column;
  background-color: white;

  button {
    width: 75px;
  }
}