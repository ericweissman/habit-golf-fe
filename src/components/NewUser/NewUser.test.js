import React from 'react';
import NewUser from './NewUser';
import { shallow } from 'enzyme'

describe('NewUser', () => {
  let wrapper
  it('should correctly match the snapshot', () => {
    wrapper = shallow(<NewUser />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should have the correct default state', () => {
    const expectedState = {
      name: '',
      email: '',
      password: '',
      confirmPassword: '',
    }
    expect(wrapper.state()).toEqual(expectedState)
  })

  it('should correctly update state based on input', () => {
    const mockEvent1 = {target: {name: 'name', value: 'Eric' }}
    const mockEvent2 = {target: {name: 'email', value: 'eric@geocities.com' }}
    const mockEvent3 = {target: {name: 'password', value: 'abc' }}
    const mockEvent4 = {target: {name: 'confirmPassword', value: 'abc' }}
    const initialState = {
      name: '',
      email: '',
      password: '',
      confirmPassword: '',
    }
    const expectedState = {
      name: 'Eric',
      email: 'eric@geocities.com',
      password: 'abc',
      confirmPassword: 'abc',
    }

    wrapper.setState(initialState)
    wrapper.instance().handleChange(mockEvent1);
    wrapper.instance().handleChange(mockEvent2);
    wrapper.instance().handleChange(mockEvent3);
    wrapper.instance().handleChange(mockEvent4);

    expect(wrapper.state()).toEqual(expectedState)
  })
})
