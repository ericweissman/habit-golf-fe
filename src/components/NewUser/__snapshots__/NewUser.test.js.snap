// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewUser should correctly match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewUser />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
          First Name:
          <input
            name="firstName"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>,
        <label>
          Last Name:
          <input
            name="lastName"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>,
        <label>
          email:
          <input
            name="email"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>,
        <label>
          Password:
          <input
            name="password"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>,
        <label>
          Confirm Password:
          <input
            name="confirmPassword"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>,
      ],
      "className": "new-user-form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "First Name:",
            <input
              name="firstName"
              onChange={[Function]}
              type="text"
              value=""
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          "First Name:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "firstName",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Last Name:",
            <input
              name="lastName"
              onChange={[Function]}
              type="text"
              value=""
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Last Name:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "lastName",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "email:",
            <input
              name="email"
              onChange={[Function]}
              type="text"
              value=""
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          "email:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Password:",
            <input
              name="password"
              onChange={[Function]}
              type="text"
              value=""
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Password:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "password",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Confirm Password:",
            <input
              name="confirmPassword"
              onChange={[Function]}
              type="text"
              value=""
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Confirm Password:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "confirmPassword",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            First Name:
            <input
              name="firstName"
              onChange={[Function]}
              type="text"
              value=""
            />
          </label>,
          <label>
            Last Name:
            <input
              name="lastName"
              onChange={[Function]}
              type="text"
              value=""
            />
          </label>,
          <label>
            email:
            <input
              name="email"
              onChange={[Function]}
              type="text"
              value=""
            />
          </label>,
          <label>
            Password:
            <input
              name="password"
              onChange={[Function]}
              type="text"
              value=""
            />
          </label>,
          <label>
            Confirm Password:
            <input
              name="confirmPassword"
              onChange={[Function]}
              type="text"
              value=""
            />
          </label>,
        ],
        "className": "new-user-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "First Name:",
              <input
                name="firstName"
                onChange={[Function]}
                type="text"
                value=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "First Name:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "firstName",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Last Name:",
              <input
                name="lastName"
                onChange={[Function]}
                type="text"
                value=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Last Name:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "lastName",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "email:",
              <input
                name="email"
                onChange={[Function]}
                type="text"
                value=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "email:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Password:",
              <input
                name="password"
                onChange={[Function]}
                type="text"
                value=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Password:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Confirm Password:",
              <input
                name="confirmPassword"
                onChange={[Function]}
                type="text"
                value=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Confirm Password:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "confirmPassword",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
